### PR TITLE
chore: setting up pre-commit hooks

### LINKED ISSUE #77: Add pre-commit hook for lint and formatting
Since this project recently added linting and formatting I think it would be nice to add a pre-commit hook.

https://prettier.io/docs/en/precommit.html

### PR DESCRIPTION
**Pull request by a2937**
I found the information on https://www.inkoop.io/blog/setup-eslint-for-react-with-prettier-pre-commit-and-vscode/

Hopefully it'll work nicely.

Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #77

Pre-commit was intimidating but pretty-quick seems pretty easy? I know it's not exactly what you want , but it could work maybe?

### CODE REVIEW

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
I tested this locally by temporarily disabling my Prettier config and I was still able push up a commit where I changed the formatting to the button component. 

I am not able to get the command to run before pushing the commit.
Maybe there is an addition configuration or change needed for pretty-quick
https://github.com/azz/pretty-quick

File: src/components/Button.tsx [hunk 1]
```diff
@@ -9,8 +9,12 @@ interface ButtonProps {
...
 
 const Button: React.FC<ButtonProps> = (ButtonProps) => {
-  const getButtonClasses = useMemo(() => {
-    let classes = "btn-default";
+  const getButtonClasses = useMemo(
+
+
+
+    () => {
+      let classes = "btn-default";
```
Comment (jdwilkin4) [inline src/components/Button.tsx:17]: I made changes to this button file to test out the pre commit hook.

You can use this file to test out your pre commit hook changes.
When successful, the pre commit hook will format it even if the user does not have es lint or prettier installed locally on their code editor. 

I also used [Option 1 ](https://prettier.io/docs/en/precommit.html) for my forked copy and recorded the pre commit results in the terminal. 

https://user-images.githubusercontent.com/67210629/177216862-41da756d-3df8-4268-a350-f1b79c75d84f.mov

Hope that helps!

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
I ran through [the pretty-quick instructions](https://prettier.io/docs/en/precommit.html#option-2-pretty-quickhttpsgithubcomazzpretty-quick) and found some things missing in your repo.

It looks like you are missing a folder in the husky folder.
When I ran the `npx husky-init` command it provided me with this folder structure
Screen Shot 2022-07-09 at 11 28 29 AM

I think that missing husky file is the problem.
I would rerun that `npx husky-init` command to make sure you have that folder in your repo. 
Then you can test your changes by poorly formatting the `Button.tsx` like I did earlier in this repo. 

Then stage and commit your changes. 
When everything is working correctly you should see this output and the Button.tsx file will be correctly formatted. 
Screen Shot 2022-07-09 at 11 35 31 AM

Hope that helps! 👍

**Comment by a2937** [comment]
Vs code screenshot

I don't think the folder will actually ever be added because the git-ignore in the folder will keep it from being pushed. Secondly when I re-run the init; the contents of my pre-commit file get changed from ```npx pretty-quick --stage``` to ```npm test```. I think I chose the right format command. However I will be more than happy to look into again after a test push.

**Comment by a2937** [comment]
PowerShell screenshot 

Another big issue I've ran into it is that I can't use Github Desktop to commit. Something about a command not found.

**Comment by a2937** [comment]
And here is the Github Desktop issue. 

Github Desktop screenshot

**Review by jdwilkin4 — APPROVED** [review]
LGTM! 👍

**Comment by jdwilkin4** [comment]
> I don't think the folder will actually ever be added because the git-ignore in the folder will keep it from being pushed.

That's true. Originally I thought it wasn't working on your end because you still had the unformatted `Button.tsx` file in your PR. But your latest push formatted that and everything looks good now. 👍

**Pull request merged**